src/elf/mod.rs

```diff
@@ -277,7 +277,15 @@ if_sylvan! {
 
             let section_headers = SectionHeader::parse(bytes, header.e_shoff as usize, header.e_shnum as usize, ctx)?;
 
-            let get_strtab = |section_headers: &[SectionHeader], section_idx: usize| {
+            let get_strtab = |section_headers: &[SectionHeader], mut section_idx: usize| {
+                if section_idx == section_header::SHN_XINDEX as usize {
+                    if section_headers.is_empty() {
+                        // FIXME: warn! here
+                        return Ok(Strtab::default())
+                    }
+                    section_idx = section_headers[0].sh_link as usize;
```

Comment:
just curious (can't remember why) it's always the first section header's sh_link that is used?

Reply:
It seems to. I actually looked what Gimli was doing, but [the ELF docs at Oracle](https://docs.oracle.com/cd/E23824_01/html/819-0690/chapter6-43405.html#scrolltoc) says:  

>If the section name string table section index is greater than or equal to SHN_LORESERVE (0xff00), this member has the value SHN_XINDEX (0xffff) and the actual index of the section name string table section is contained in the sh_link field of the section header at index 0. Otherwise, the sh_link member of the initial section header entry contains the value zero. See [Table 12-6](https://docs.oracle.com/cd/E23824_01/html/819-0690/chapter6-94076.html#chapter6-12754) and [Table 12-7](https://docs.oracle.com/cd/E23824_01/html/819-0690/chapter6-94076.html#gavyo).